src/ray/raylet/worker_pool.cc

```diff
@@ -971,7 +954,9 @@ void WorkerPool::TryKillingIdleWorkers() {
     auto worker_startup_token = idle_worker->GetStartupToken();
     auto &worker_state = GetStateForLanguage(idle_worker->GetLanguage());
 
-    if (worker_state.starting_worker_processes.count(worker_startup_token) > 0) {
+    auto it = worker_state.worker_processes.find(worker_startup_token);
+    if (it != worker_state.worker_processes.end() &&
+        it->second.num_starting_workers != 0) {
```

Comment:
Can `num_starting_workers ` be negative?

Reply:
I think it can't. But I also could change this code to `it->second.num_starting_workers > 0`.